src/algebra/group/anti_hom.lean

```diff
@@ -12,7 +12,7 @@ variables {α : Type*} {β : Type*} [group α] [group β]
 /-- Predicate for group anti-homomorphism, or a homomorphism
   into the opposite group. -/
 class is_group_anti_hom (f : α → β) : Prop :=
-(map_mul : ∀ a b : α, f (a * b) = f b * f a)
+(map_mul [] : ∀ a b : α, f (a * b) = f b * f a)
```

Comment:
I see lots of added `[]`. What have changed in `3.8.0`? I just want to know when I should put `[]` in the new code.

Reply:
This is due to https://github.com/leanprover-community/lean/pull/175

Short story: you've probably noticed that `list.cons` has `α` as an implicit argument, while `list` has `α` as an explicit argument.  We've changed default strategy to decide which arguments should be implicit.  Previously the rule was: anything that appears in the type of an explicit argument becomes implicit.  The new default rule is: in addition, anything that appears in the return type becomes implicit.  (That is, `list.nil` has `α` as an implicit argument in the new rules but not in the old rules.)  You can use `[]` to get the old behavior back.

Long story: the unifier has a heuristic that unifies *implicit* arguments with a higher transparency.  That is, when unifying implicit arguments, it unfolds all definitions unless they are tagged `@[irreducible]`.  Hence it is important that types are implicit arguments (see https://github.com/leanprover-community/lean/pull/169).  Therefore the new default is the right choice for `list.nil`, `has_zero.zero`, etc.  It might also become the default in Lean 4.

Reply:
Previously, we've marked lots of constructors as `{}` manually: https://github.com/leanprover-community/mathlib/pull/2361/commits/c0231e675dc27bb65281bdc93a87f96fc6a801df

Reply:
Thank you for the explanation!